### PR TITLE
refactor: use TransactionStatus component

### PR DESCRIPTION
**Pull request by 0xApotheosis**
## Description

Add a `TransactionStatus` component to reduce duplication across transaction rendering components.

## Notice

Before submitting a pull request, please make sure you have answered the following:

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [x] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

Addresses comment https://github.com/shapeshift/web/pull/1076#discussion_r817016617

## Risk

N/A

## Testing

`yarn dev` and ensure transactions still display a status.

## Screenshots (if applicable)

N/A

### CODE REVIEW

**Comment by 0xApotheosis** [comment]
This one's for you, @gomesalexandre!

File: src/components/Transactions/TransactionStatus.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      case chainAdapters.TxStatus.Pending:
+        return (
+          <>
+            <Tag colorScheme='blue' size='lg'>
+              <CircularProgress mr={2} size='5' />
+              <Text translation='transactionRow.pending' />
+            </Tag>
+          </>
+        )
+      default:
```
Comment (gomesalexandre) [inline src/components/Transactions/TransactionStatus.tsx:29]: Why not be exhaustive and go with `case chainAdapters.TxStatus.Failed` here?

**Review by gomesalexandre — APPROVED** [review]
LGTM 🥇 I couldn't find any usage of the previous, unabstracted tx status nodes. 

Tested it and saw no regressions on send, receive and contract interaction transactions components.

Nice use of IIFE! with `status` as well! I was thinking of a render function we'd call but this looks actually even nicer!

File: src/components/Transactions/TransactionStatus.tsx (hunk 1 above)
Comment (0xApotheosis) [reply src/components/Transactions/TransactionStatus.tsx:30]: Yeah true, we could do a fall-through: https://github.com/shapeshift/web/pull/1094/commits/d2150eda4c5f707429e3c2dcba9d2444a640b853

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**